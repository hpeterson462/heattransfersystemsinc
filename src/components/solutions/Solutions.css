.sol-card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  transform: translateY(20vh);
  background-color:#F9FAFB;
}

.sol-card-container.is-visible {
  opacity: 1;
  transform: none;
  visibility: visible;
}

.sol-wrapper {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  margin: 30px;
  max-width: 80%;
}

.sol-card {
  position: relative;
  margin: 30px;
  align-items: center;
  width: 300px;
  height: 300px;
  border-radius: 2px;
  box-shadow: 0 0 5px 2px rgba(0, 0, 0, 0.3);
  background-color: #F9FAFB;
  border: #003366 2px solid;
}

.sol-name {
  text-align: center;
  margin: auto;
  padding: 5px;
  width: 300px;
  border-radius: 2px;
  background-color: #F9FAFB;
  font-size: 1.2rem;
  letter-spacing: 0.2rem;
}

.sol-img {
  width: 300px;
  height: 300px;
  border-radius: 2px;
  transition: opacity .5s ease-in-out;
}

.sol-description {
  position: absolute;
  height: 160px;
  width: 160px;
  opacity: 0;
  top: 0;
  left: 0;
  padding: 70px;
  border-radius: 2px;
  transition: 0.8s;
  background-color: rgba(246, 141, 11, 0.5);
}

.sol-description:hover {
  cursor: pointer;
  border-radius: 2px;
  opacity: 1;
  transition: opacity .5s ease-in-out;
}

.sol-p {
  text-align: left;
  margin: auto;
  padding: 10px;
  border-radius: 2px;
  font-family: 'Roboto', sans-serif;
  line-height: 1.3rem;  
  background-color:#F9FAFB;
}
