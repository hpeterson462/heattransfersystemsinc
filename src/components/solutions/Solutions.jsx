import React from 'react';
import { useScrollAnimation } from '../../hooks/scrollAnimation';
import burnerEquipment from '../../assets/burnerEquipment.jpg';
import controlPanel from '../../assets/controlPanel.jpg';
import turnKeySystems from '../../assets/turnKeySystems.jpg';
import gasTrain from '../../assets/gasTrain.jpg';
import './Solutions.css';

export default function Solutions() {

  const { isVisible, animationRef } = useScrollAnimation();

  return (
    <>
      <section>
        <h1 id="solutions">
          Solutions
        </h1>

        <h2>
          We build everything with the highest quality components and detail to deliver safe and reliable equipment that you can count on for years to come.
        </h2>
      </section>

      <div className={`sol-card-container ${isVisible ? 'is-visible' : ''}`} ref={animationRef}>
        <div className="sol-wrapper">
          <div>
            <div className="sol-name">Burner Equipment</div>
            <div className="sol-card">
              <img
                className="sol-img"
                src={burnerEquipment}
                alt="industrial heating system"
              />
              <div className="sol-description">
                <p className="sol-p">
                  <b>Gas & oil burners</b>, <b>recuperators</b>, and <b>heat exchangers</b> for standard to low emissions applications in a variety of industries.
              </p>
              </div>
            </div>
          </div>

          <div>
            <div className="sol-name">Gas Trains</div>
            <div className="sol-card">
              <img
                className="sol-img"
                src={gasTrain}
                alt="gas train"
              />
              <div className="sol-description">
                <p className="sol-p">
                  <b>Custom gas trains</b> built to meet <b>NFPA</b> requirements for <b>natural & propane</b> gas tailored to the unique needs of our customers.
              </p>
              </div>
            </div>
          </div>

          <div>
            <div className="sol-name">Control Panels</div>
            <div className="sol-card">
              <img
                className="sol-img"
                src={controlPanel}
                alt="industrial heating system panel"
              />
              <div className="sol-description">
                <p className="sol-p">
                  From <b>single-burner</b> to <b>complex PLC</b> based controls, we provide custom control panels for each system.
              </p>
              </div>
            </div>
          </div>

          <div>
            <div className="sol-name">Turn-Key Systems</div>
            <div className="sol-card">
              <img
                className="sol-img"
                src={turnKeySystems}
                alt="heat exchanger"
              />
              <div className="sol-description">
                <p className="sol-p">
                  From tear down of old equipment to the <b>design & installation</b> of new systems, we provide a <b>complete package</b> in start-up training.
              </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  )
}
