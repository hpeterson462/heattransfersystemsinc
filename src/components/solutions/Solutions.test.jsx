import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Solutions from './Solutions';

describe('Solutions component', () => {
  afterEach(() => cleanup());

  it('renders Solutions component', () => {
    const { asFragment } = render(<MemoryRouter><Solutions /></MemoryRouter>);

    expect(asFragment()).toMatchSnapshot();
  });
});
