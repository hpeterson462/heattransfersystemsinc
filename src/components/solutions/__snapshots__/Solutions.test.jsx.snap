// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Solutions component renders Solutions component 1`] = `
<DocumentFragment>
  <section>
    <h1
      id="solutions"
    >
      Solutions
    </h1>
    <h2>
      We build everything with the highest quality components and detail to deliver safe and reliable equipment that you can count on for years to come.
    </h2>
  </section>
  <div
    class="sol-card-container is-visible"
  >
    <div
      class="sol-wrapper"
    >
      <div>
        <div
          class="sol-name"
        >
          Burner Equipment
        </div>
        <div
          class="sol-card"
        >
          <img
            alt="industrial heating system"
            class="sol-img"
            src="burnerEquipment.jpg"
          />
          <div
            class="sol-description"
          >
            <p
              class="sol-p"
            >
              <b>
                Gas & oil burners
              </b>
              , 
              <b>
                recuperators
              </b>
              , and 
              <b>
                heat exchangers
              </b>
               for standard to low emissions applications in a variety of industries.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="sol-name"
        >
          Gas Trains
        </div>
        <div
          class="sol-card"
        >
          <img
            alt="gas train"
            class="sol-img"
            src="gasTrain.jpg"
          />
          <div
            class="sol-description"
          >
            <p
              class="sol-p"
            >
              <b>
                Custom gas trains
              </b>
               built to meet 
              <b>
                NFPA
              </b>
               requirements for 
              <b>
                natural & propane
              </b>
               gas tailored to the unique needs of our customers.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="sol-name"
        >
          Control Panels
        </div>
        <div
          class="sol-card"
        >
          <img
            alt="industrial heating system panel"
            class="sol-img"
            src="controlPanel.jpg"
          />
          <div
            class="sol-description"
          >
            <p
              class="sol-p"
            >
              From 
              <b>
                single-burner
              </b>
               to 
              <b>
                complex PLC
              </b>
               based controls, we provide custom control panels for each system.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="sol-name"
        >
          Turn-Key Systems
        </div>
        <div
          class="sol-card"
        >
          <img
            alt="heat exchanger"
            class="sol-img"
            src="turnKeySystems.jpg"
          />
          <div
            class="sol-description"
          >
            <p
              class="sol-p"
            >
              From tear down of old equipment to the 
              <b>
                design & installation
              </b>
               of new systems, we provide a 
              <b>
                complete package
              </b>
               in start-up training.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
